tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -22,27 +22,63 @@ def decode_name_with_port(input_model: InputModel, node_name: str, framework="")
     found_nodes = []
     found_node_names = []
 
-    node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        found_node_names.append('Tensor:' + node_name)
-        found_nodes.append(node)
+    def try_get_tensor(node_name):
+        tensor = input_model.get_place_by_tensor_name(node_name)
+        if tensor:
+            return tensor
+        if node_name.count(":"):
+            regexp_pre = r'(\d+):(.+)'
+            match_pre = re.search(regexp_pre, node_name)
+            if match_pre:
+                port, name = node_name.split(":")
+                node = input_model.get_place_by_operation_name(name)
+                if node:
+                    tensor = node.get_source_tensor(input_port_index=int(port))
+                    if tensor:
+                        return tensor
+                tensor = input_model.get_place_by_tensor_name(name)
```

Comment:
Is my understanding correct that if we have tensor with name 'myTensor', then `try_get_tensor("myTensor:12345")` will always return this tensor? Is it expected behavior?

Reply:
My understanding is that we either expect that there is tensor with name "myTensor:12345" or operation with name "myTensor" and we want its output port 12345. At the moment if we have tensor with name "myTensor" then it will no be returned if we use ```try_get_tensor("myTensor:12345")``` . The function will return tensor only if there is an tensor with name " myTensor:12345" or operation with name "myTensor" and output port 12345. Is it expected behaviour?